e2e-testing/readme.md

```diff
@@ -0,0 +1,164 @@
+
+Below enlisted are few **guidelines** that should be kept in mind while designing the test cases for performing E2E testing:
+
+- Test cases should be designed from the end user’s perspective.
+- Should focus on testing some existing features of the system.
+- Multiple scenarios should be considered for creating multiple test cases.
+- Different sets of test cases should be created to focus on multiple scenarios of the system.
+
+## E2E Testing Design Framework
+
+![E2E Testing Design Framework](./images/e2e-testing-design-framework.jpg)
+
+We will look into all the 3 categories one by one:
+
+### User Functions
+
+Following actions should be performed as a part of building user functions:
+
+- List user initiated functions of the software systems and their interconnected sub-systems.
+- For any function, keep track of the actions performed as well as Input and Output data.
+- Find the relations, if any between different Users functions.
+- Find out the nature of different user functions i.e. if they are independent or are reusable.
+
+### Conditions
+
+Following activities should be performed as a part of building conditions based on user functions:
+
+- For each and every user functions, a set of conditions should be prepared.
+- Timing, data conditions and other factors that affect user functions can be considered as parameters.
+
+### Test Cases
+
+Following factors should be considered for building test cases:
+
+- For every scenario, one or more test cases should be created to test each and every functionality of the user functions.
+- Every single condition should be enlisted as a separate test case.
+
+## E2E Testing Progress Tracking
+
+Moving to the next important activities involved in every E2E testing:
+
+- **Status of Test case preparation**: This can be tracked in the form of a dashboard percentage completion to represent the progress of the planned test cases that are under preparation.
+- **Weekly tracking of Test progress**: This includes week wise representation of the test cases execution progress. It can be reflected through percentage representation for a pass, fail, executed, not executed, invalid, etc cases.
+- **Status and detailed report for Defects**: The Status report should be prepared on a daily basis to show the test case execution status as well as defects found and logged as per their severity. Weekly, the percentage of the open and closed defects should be calculated. Also, based on defect severity and priority, defects status should be tracked on a weekly basis.
+- **Test environment**: This keeps a track of the test environment time duration allotted as well as the test environment time actually used while performing E2E testing.
+
+## E2E Testing Frameworks and Tools
```

Comment:
How about [Robot Framework](https://robotframework.org/)?

Reply:
Good point. I've added the section on Robot Framework.